src/plugins/intel_gpu/src/plugin/transformations/remove_unnecessary_pad.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "remove_unnecessary_pad.hpp"
+#include <memory>
+
+#include "openvino/op/constant.hpp"
+#include "openvino/op/pad.hpp"
+#include "openvino/op/max_pool.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/op/pattern.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+RemoveUnnecessaryPad::RemoveUnnecessaryPad() {
+    using namespace ov::op;
+    using namespace ov::pass;
+    auto pads_begin = pattern::wrap_type<v0::Constant>();
+    auto pads_end = pattern::wrap_type<v0::Constant>();
+    auto arg_pad_value = pattern::wrap_type<v0::Constant>();
+    auto pad = pattern::wrap_type<v12::Pad>({ov::pass::pattern::any_input(),
+                                                               pads_begin, pads_end, arg_pad_value});
+    auto maxpool_v1 = pattern::wrap_type<v1::MaxPool>({pad});
+    auto maxpool_v8 = pattern::wrap_type<v8::MaxPool>({pad});
+    auto maxpool_v1_or_v8 = std::make_shared<pattern::op::Or>(OutputVector{maxpool_v1, maxpool_v8});
```

Comment:
Is it legal at all? You check that Pad adds zero values and assume that it doesn't impact max value, but if maxpool input has all negative values, then max will become 0.

Reply:
When pads_begin and pads_end are zero, the output size of Pad has not changed so that arg_pad_value is never used. So It cannot affect the values of output.  Am I misunderstanding something?

Reply:
Oh, I see. So then I don't understand another thing:
1. If the transformation removes unnecessary pad, why do you need maxpool in the pattern at all?
2. We have common transform for pad elimination [here ](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp#L348)  - why it didn't cover the target scenario?

Reply:
1. You are right. I will replace with EliminatePad pass.
2. This pad is inserted when downgrading v14::MaxPool having ceil_pytorch rounding type to v8::MaxPool. [code](https://github.com/openvinotoolkit/openvino/blob/d481f34744bfecd7144550f35e515c5763db658c/src/common/transformations/src/transformations/op_conversions/convert_maxpool_downgrade.cpp#L71) After this downgrading, pad elimination is not called.